cmd/test_util.go

```diff
@@ -112,3 +116,29 @@ func assertEqualYaml(fileToFix string, fileFixed string, auditFunc func(resource
 	assert.Nil(err)
 	assert.Equal(correctlyFixedResources[0], fixedResource)
 }
+
+// WriteToTmpFile writes a single resource to a tmpfile, you are responsible
+// for closing the file afterwards, that's why the function returns the file
+// name.
+func WriteToTmpFile(decode Resource) (string, error) {
+	info, _ := k8sRuntime.SerializerInfoForMediaType(scheme.Codecs.SupportedMediaTypes(), "application/yaml")
+	groupVersion := schema.GroupVersion{Group: decode.GetObjectKind().GroupVersionKind().Group, Version: decode.GetObjectKind().GroupVersionKind().Version}
+	encoder := scheme.Codecs.EncoderForVersion(info.Serializer, groupVersion)
+	yaml, err := k8sRuntime.Encode(encoder, decode)
+	if err != nil {
+		return "", err
+	}
+	tmpfile, err := ioutil.TempFile("", "kubeaudit-test-yaml")
+	if err != nil {
+		return "", err
+	}
+	_, err = tmpfile.Write(yaml)
+	if err != nil {
+		return "", err
+	}
+	err = tmpfile.Close()
+	if err != nil {
+		return "", err
```

Comment:
Why return an empty string on failed `.Close?` or write calls? Makes sense to me on open but in theory the filename might be useful to output on error by the caller?

Reply:
There are only 2 cases in which an error can occur for a `file.Close()`, iirc. And correct me if I am wrong:
1. The file handle is `nil`. This would mean something super weird has happened because no other program should have the file handle because we just created it and this is single threaded so I don't see how it could have been `nil`.
1. The file has already be closed. If this is the case you already get an error that contains the filename in its error message.

So I cannot see the first one happen and for the second one you already get an informative error message.

Reply:
👍